ZelBack/src/services/fluxNetworkHelper.js

```diff
@@ -254,16 +255,36 @@ async function checkAppAvailability(req, res) {
   });
   req.on('end', async () => {
     try {
+      const authorized = await verificationHelper.verifyPrivilege('adminandfluxteam', req);
+
       const processedBody = serviceHelper.ensureObject(body);
 
-      const { ip, ports, appname } = processedBody;
+      const {
+        ip, ports, appname, pubKey, signature,
+      } = processedBody;
+
+      // pubkey of the message has to be on the list
+      const zl = await fluxCommunicationUtils.deterministicFluxList(pubKey); // this itself is sufficient.
+      const node = zl.find((key) => key.pubkey === pubKey); // another check in case sufficient check failed on daemon level
+      const dataToVerify = processedBody;
+      delete dataToVerify.signature;
+      const messageToVerify = JSON.stringify(dataToVerify);
+      const verified = verificationHelper.verifyMessage(messageToVerify, pubKey, signature);
+      if ((verified !== true || !node) && authorized !== true) {
+        log.error('Unable to verify request authenticity');
+        // throw new Error('Unable to verify request authenticity');
```

Comment:
Do we want to do a `return` here if authenticity is not verified or let it keep going?

Reply:
Currently keep it going. Once nodes update we will most likely uncomment this throw